packages/form-field/src/styles.tsx

```diff
@@ -0,0 +1,84 @@
+import styled, {css} from 'styled-components'
+import {Label as PureLabel, Paragraph, getTextStyle} from '@smashing/typography'
+import {Alert as PureAlert, StyledAlert} from '@smashing/alert'
+import {FormFieldLabelAppearance, FormFieldAlertAppearance} from './types'
+
+export const Description = styled(Paragraph)`
+  margin: 0 0 ${_ => _.theme.spacing.xxxs};
+`
+
+export const Hint = styled(Paragraph)`
+  margin: ${_ => _.theme.spacing.xxxs} 0 0;
+`
+
+interface LabelProps {
+  hasDescription: boolean
+  appearance?: FormFieldLabelAppearance
+}
+
+export const Label = styled(PureLabel)<LabelProps>`
+  ${_ =>
+    _.appearance === 'block' &&
+    css`
+      margin-bottom: ${_.hasDescription ? 0 : _.theme.spacing.xxxs};
+    `}
+
+  ${_ => _.appearance === 'inline' && css``}
+
+  ${_ =>
+    _.appearance === 'overlay' &&
+    css`
+      position: absolute;
+      top: -14px;
```

Comment:
This works only for certain variant's font size, we can calculate from used variant if provided by theme/to the component